google/cloud/bigquery/schema.py

```diff
@@ -886,3 +903,131 @@ def from_api_repr(cls, resource: dict) -> SerDeInfo:
+        use of a '.data' attribute. The decision was made to have both of these attributes point to
+        the same key "fields" in the '_properties' dictionary. Thus '.data' is an alias
+        for '_fields'.
+
+        Args:
+            value (Optional[List[Any]]): A list of schema fields, can be set to None.
+        """
+        value = _isinstance_or_raise(value, list, none_allowed=True)
+        value = _build_schema_resource(value)
+        self._properties["fields"] = value
+
+    @property
+    def data(self) -> Any:
+        return self._properties.get("fields")
+
+    @data.setter
+    def data(self, value) -> None:
+        """
+        Sets the list of schema fields.
+
+        NOTE: In the API representation the 'fields' key points to a sequence of schema fields.
+        To maintain a similarity in names, the 'Schema._fields' attribute points to the
+        '_properties["fields"]' key. Schema class is superclassed by UserList, which requires the
+        use of a '.data' attribute. The decision was made to have both of these attributes point to
+        the same key "fields" in the '_properties' dictionary. Thus '.data' is an alias
+        for '_fields'.
+
+        Args:
+            value (Optional[List[Any]]): A list of schema fields, can be set to None.
+        """
+
+        value = _isinstance_or_raise(value, list, none_allowed=True)
+        value = _build_schema_resource(value)
+        self._properties["fields"] = value
+
+    def __str__(self) -> str:
+        return f"Schema({self._fields}, {self.foreign_type_info})"
+
+    def __repr__(self) -> str:
+        return f"Schema({self._fields!r}, {self.foreign_type_info!r})"
+
+    def to_api_repr(self) -> Dict[str, Any]:
+        """Build an API representation of this object.
+
+        Returns:
+            Dict[str, Any]:
+                A dictionary in the format used by the BigQuery API.
+                If the schema contains SchemaField objects, the fields are
+                also converted to their API representations.
+        """
+        # If this is a RECORD type, then sub-fields are also included,
+        # add this to the serialized representation.
+        answer = self._properties.copy()
+        if self._fields == []:
+            return answer
+        schemafields = any([isinstance(f, SchemaField) for f in self._fields])
+        if schemafields:
+            answer["fields"] = [f.to_api_repr() for f in self._fields]
+        return answer
+
+    @classmethod
+    def from_api_repr(cls, resource: Dict[str, Any]) -> "Schema":
+        """Factory: constructs an instance of the class (cls)
+        given its API representation.
+
+        Args:
+            resource (Dict[str, Any]):
+                API representation of the object to be instantiated.
+
+        Returns:
+            An instance of the class initialized with data from 'resource'.
+        """
+        config = cls([])
+        config._properties = copy.copy(resource)
```

Comment:
With using `UserList`, you will need to pull out `fields` and pass them into the constructor.

```suggestion
        resource = copy.copy(resource)
        fields = resource.pop("fields", [])
        config = cls(fields)
        config._properties = resource
```

This makes sure `.data` has all `SchemaField` objects while preserving the future-proof ability to add `_properties` (other than `fields`).